platform/logging/LoggingConfig.ts

```diff
@@ -0,0 +1,166 @@
+import { Schema, typeOfSchema } from '../types';
+import { LogLevelId } from './LogLevel';
+import { Appenders, AppenderConfigType } from './appenders/Appenders';
+
+/**
+ * Separator string that used within nested context name (eg. plugins::pid).
+ */
+const CONTEXT_SEPARATOR = '::';
+
+/**
+ * Name of the `root` context that should always be defined.
+ */
+const ROOT_CONTEXT_NAME = 'root';
+
+const createLoggerSchema = (schema: Schema) => {
+  const { literal, object, string, arrayOf, oneOf } = schema;
+
+  return object({
+    level: oneOf(
+      [
+        literal('all'),
+        literal('fatal'),
+        literal('error'),
+        literal('warn'),
+        literal('info'),
+        literal('debug'),
+        literal('trace'),
+        literal('off')
+      ],
+      {
+        defaultValue: 'info'
+      }
+    ),
+    appenders: arrayOf(string(), { defaultValue: [] })
+  });
+};
+
+const createLoggingSchema = (schema: Schema) => {
+  const { object, string, mapOf } = schema;
+
+  return object({
+    appenders: mapOf(string(), Appenders.createConfigSchema(schema), {
+      defaultValue: new Map([
+        [
+          'console',
+          {
+            kind: 'console',
+            layout: {
+              kind: 'pattern',
+              highlight: true
+            }
+          }
+        ]
+      ])
+    }),
+    loggers: mapOf(string(), createLoggerSchema(schema), {
+      defaultValue: new Map([
+        ['root', { appenders: ['console'], level: 'info' as LogLevelId }]
+      ]),
+      validate(value) {
+        const rootLogger = value.get(ROOT_CONTEXT_NAME);
```

Comment:
Having `root` in the common `loggers` config and its appender in common `appenders` config makes code much easier, but config becomes harder to tune: e.g. in case you want to add a new logger you'll have to define `root` logger explicitly, same for new appender - you'll have to define appender that `root` logger uses explicitly as well. I'd definitely improve this somehow, but maybe in the next PR. What do you think? Do you have any ideas?